std/file.d

```diff
@@ -188,7 +226,15 @@ Returns: Untyped array of bytes _read.
 
 Throws: $(D FileException) on error.
  */
-version (Posix) void[] read(in char[] name, size_t upTo = size_t.max) @trusted
+version (Posix) void[] read(R)(R name, size_t upTo = size_t.max)
+{
+    static if (isNarrowString!R && is(Unqual!(ElementEncodingType!R) == char))
+        return readImpl(name, name.tempCString(), upTo);
+    else
+        return readImpl(null, name.tempCString(), upTo);
+}
+
+version (Posix) private void[] readImpl(const(char)[] name, const(char)* namez, size_t upTo = size_t.max) @trusted
```

Comment:
I wanted it outside as a non-template function so it will not get repeatedly generated.